conduit/plugins/exporters/filewriter/util.go

```diff
@@ -1,52 +1,96 @@
 package filewriter
 
 import (
-	"bytes"
 	"compress/gzip"
 	"fmt"
 	"io"
 	"os"
 	"strings"
 
 	"github.com/algorand/go-algorand-sdk/v2/encoding/json"
+	"github.com/algorand/go-algorand-sdk/v2/encoding/msgpack"
 	"github.com/algorand/go-codec/codec"
 )
 
-var prettyHandle *codec.JsonHandle
-var jsonStrictHandle *codec.JsonHandle
+// EncodingFormat enumerates the acceptable encoding formats for Conduit file-based plugins.
+type EncodingFormat byte
+
+const (
+	// MessagepackFormat indicates the file is encoded using MessagePack.
+	MessagepackFormat EncodingFormat = iota
+
+	// JSONFormat indicates the file is encoded using JSON.
+	JSONFormat
+
+	// UnrecognizedFormat indicates the file's encoding is unknown to Conduit.
+	UnrecognizedFormat
+)
+
+var jsonPrettyHandle *codec.JsonHandle
 
 func init() {
-	prettyHandle = new(codec.JsonHandle)
-	prettyHandle.ErrorIfNoField = json.CodecHandle.ErrorIfNoField
-	prettyHandle.ErrorIfNoArrayExpand = json.CodecHandle.ErrorIfNoArrayExpand
-	prettyHandle.Canonical = json.CodecHandle.Canonical
-	prettyHandle.RecursiveEmptyCheck = json.CodecHandle.RecursiveEmptyCheck
-	prettyHandle.Indent = json.CodecHandle.Indent
-	prettyHandle.HTMLCharsAsIs = json.CodecHandle.HTMLCharsAsIs
-	prettyHandle.MapKeyAsString = true
-	prettyHandle.Indent = 2
-
-	jsonStrictHandle = new(codec.JsonHandle)
-	jsonStrictHandle.ErrorIfNoField = prettyHandle.ErrorIfNoField
-	jsonStrictHandle.ErrorIfNoArrayExpand = prettyHandle.ErrorIfNoArrayExpand
-	jsonStrictHandle.Canonical = prettyHandle.Canonical
-	jsonStrictHandle.RecursiveEmptyCheck = prettyHandle.RecursiveEmptyCheck
-	jsonStrictHandle.Indent = prettyHandle.Indent
-	jsonStrictHandle.HTMLCharsAsIs = prettyHandle.HTMLCharsAsIs
-	jsonStrictHandle.MapKeyAsString = true
+	jsonPrettyHandle = new(codec.JsonHandle)
+	jsonPrettyHandle.ErrorIfNoField = json.CodecHandle.ErrorIfNoField
+	jsonPrettyHandle.ErrorIfNoArrayExpand = json.CodecHandle.ErrorIfNoArrayExpand
+	jsonPrettyHandle.Canonical = json.CodecHandle.Canonical
+	jsonPrettyHandle.RecursiveEmptyCheck = json.CodecHandle.RecursiveEmptyCheck
+	jsonPrettyHandle.Indent = json.CodecHandle.Indent
+	jsonPrettyHandle.HTMLCharsAsIs = json.CodecHandle.HTMLCharsAsIs
+	jsonPrettyHandle.MapKeyAsString = true
+	jsonPrettyHandle.Indent = 2
 }
 
-// EncodeJSONToFile is used to encode an object to a file. If the file ends in .gz it will be gzipped.
-func EncodeJSONToFile(filename string, v interface{}, pretty bool) error {
-	var writer io.Writer
+// ParseFilenamePattern parses a filename pattern into an EncodingFormat and gzip flag.
+func ParseFilenamePattern(pattern string) (EncodingFormat, bool, error) {
```

Comment:
Do these need to be public? There is some code shared between the filewriter and filereader, that could explain why the Encode function is public.

Reply:
It's used by by the filereader as well. I was thinking of creating a common library. But I think this is fine as is for now.